equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/util/ProfileAcceptedTypesUtil.java

```diff
@@ -0,0 +1,20 @@
+package uk.gov.justice.laa.crime.equinity.historicaldata.util;
+
+import jakarta.annotation.Nullable;
+import lombok.experimental.UtilityClass;
+import uk.gov.justice.laa.crime.equinity.historicaldata.exception.UnauthorizedUserProfileException;
+
+@UtilityClass
+public class ProfileAcceptedTypesUtil {
+    public static void checkTypeIsAcceptedByProfile(Integer requiredType, @Nullable String types) throws UnauthorizedUserProfileException {
+        if (types == null) return;
+
+        for (String profile : types.split(",")) {
+            if (profile.equals(requiredType.toString())) return;
```

Comment:
possible NPE, requiredType may be null? - need to do a null check before converting into tostring

Reply:
solved